frontend/projects/upgrade/src/app/features/dashboard/home/components/experiment-query-result/experiment-query-result.component.ts

```diff
@@ -32,23 +58,165 @@ export class ExperimentQueryResultComponent implements OnInit, OnDestroy {
     });
     this.analysisService.executeQuery(queryIds);
     this.queryResultsSub = this.analysisService.queryResult$.pipe(filter((result) => !!result)).subscribe((result) => {
-      result.map((res) => {
-        let resultData = res.result.map((data) => ({
+      // main effect graph data
+      this.populateMainEffectGraphData(result);
+
+      // interactive effect graph data
+      this.populateInteractionGraphData(result);
+    });
+  }
+  populateMainEffectGraphData(result: any[]) {
+    result.map((res) => {
```

Comment:
Please provide a type, and change to `forEach`, this map doesn't return anything.